monai/transforms/spatial/array.py

```diff
@@ -842,10 +842,12 @@ def __call__(
             scale = self.spatial_size / max(img_size)
             spatial_size_ = tuple(int(round(s * scale)) for s in img_size)
 
-        if tuple(img.shape[1:]) == spatial_size_:  # spatial shape is already the desired
-            return convert_to_tensor(img, track_meta=get_track_meta())  # type: ignore
-
         original_sp_size = img.shape[1:]
+        _mode = look_up_option(self.mode if mode is None else mode, InterpolateMode)
+        _align_corners = self.align_corners if align_corners is None else align_corners
+        if tuple(img.shape[1:]) == spatial_size_:  # spatial shape is already the desired
+            img = convert_to_tensor(img, track_meta=get_track_meta())  # type: ignore
```

Comment:
We could keep all the code in one method with `if/else` instead of an extra return and separate method:
```suggestion
            out = convert_to_tensor(img, track_meta=get_track_meta())  # type: ignore
        else:
            ...   # everything up to line 870 in this block
        # body of _post_process here
        return out 
```

Reply:
the separate method is actually helpful for the upcoming lazy resampling...